File: app/src/main/java/com/georgcantor/wallpaperapp/ui/AboutActivity.java
package com.georgcantor.wallpaperapp.ui;

import android.content.Intent;
import android.content.res.AssetManager;
import android.graphics.Typeface;
import android.net.Uri;
import android.os.Bundle;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.CardView;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.georgcantor.wallpaperapp.R;

public class AboutActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);

        Toolbar toolbar = findViewById(R.id.toolbar_layout_about);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayShowTitleEnabled(false);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        FloatingActionButton fab = findViewById(R.id.fab_about);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                sendEmail();
            }
        });

        AppBarLayout appBarLayout = findViewById(R.id.app_bar_about);
        appBarLayout.addOnOffsetChangedListener(new AppBarLayout.OnOffsetChangedListener() {

            boolean isShow = false;
            int scrollRange = -1;

            @Override
            public void onOffsetChanged(AppBarLayout appBarLayout, int verticalOffset) {
                if (scrollRange == -1) {
                    scrollRange = appBarLayout.getTotalScrollRange();
                }
                if (scrollRange + verticalOffset == 0) {
                    isShow = true;
                } else if (isShow) {
                    isShow = false;
                }
            }
        });

        AssetManager assetManager = this.getApplicationContext().getAssets();
        Typeface montserrat_regular =
                Typeface.createFromAsset(assetManager, "fonts/Montserrat-Regular.ttf");

        TextView aboutHeaderAppDescription = findViewById(R.id.about_header_app_description);
        aboutHeaderAppDescription.setTypeface(montserrat_regular);

        TextView cardInfo = findViewById(R.id.tv_card_info);
        cardInfo.setTypeface(montserrat_regular);

        TextView madeWithLove = findViewById(R.id.tv_made_with_love);
        madeWithLove.setTypeface(montserrat_regular);

        TextView tryMyApp = findViewById(R.id.tv_try_my_new_app);
        tryMyApp.setTypeface(montserrat_regular);

        CardView cardViewInfo = findViewById(R.id.card_info);
        CardView cardViewNewApp = findViewById(R.id.card_new_app);

        cardViewInfo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent browserIntent = new Intent(Intent.ACTION_VIEW,
                        Uri.parse("https://play.google.com/store/apps/dev?id=5242637664196553916&hl=en"));
                startActivity(browserIntent);
            }
        });

        cardViewNewApp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent browserIntent = new Intent(Intent.ACTION_VIEW,
                        Uri.parse("https://play.google.com/store/apps/details?id=com.georgcantor.vipnews&hl=en"));
                startActivity(browserIntent);
            }
        });
    }

    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    protected void sendEmail() {
        Log.i("Send email", "");
        String[] TO = {"cupsman1986@gmail.com"};
        Intent emailIntent = new Intent(Intent.ACTION_SEND);

        emailIntent.setData(Uri.parse("mailto:"));
        emailIntent.setType("text/plain");
        emailIntent.putExtra(Intent.EXTRA_EMAIL, TO);
        emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Your subject");
        emailIntent.putExtra(Intent.EXTRA_TEXT, "Email message goes here");

        try {
            startActivity(Intent.createChooser(emailIntent, "Send mail..."));
            finish();
            Log.i("Finished sending email", "");
        } catch (android.content.ActivityNotFoundException ex) {
            Toast.makeText(AboutActivity.this,
                    "There is no email client installed.", Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        overridePendingTransition(R.anim.pull_in_left, R.anim.push_out_right);
    }
}

File: app/src/main/java/com/georgcantor/wallpaperapp/ui/AboutActivity.kt
package com.georgcantor.wallpaperapp.ui

import android.content.Intent
import android.graphics.Typeface
import android.net.Uri
import android.os.Bundle
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.georgcantor.wallpaperapp.R
import com.google.android.material.appbar.AppBarLayout
import kotlinx.android.synthetic.main.about_header.*
import kotlinx.android.synthetic.main.activity_about.*
import kotlinx.android.synthetic.main.card_info.*
import kotlinx.android.synthetic.main.card_music_app.*
import kotlinx.android.synthetic.main.card_new_app.*
import kotlinx.android.synthetic.main.content_about.*

class AboutActivity : AppCompatActivity() {

    companion object {
        private const val DEV_URL = "https://play.google.com/store/apps/dev?id=5242637664196553916&hl=en"
        private const val NEWS_URL = "https://play.google.com/store/apps/details?id=com.georgcantor.vipnews&hl=en"
        private const val MUSIC_URL = "https://play.google.com/store/apps/details?id=com.georgcantor.player"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)

        setSupportActionBar(aboutToolbar)
        supportActionBar?.setDisplayShowTitleEnabled(false)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        aboutFab.setOnClickListener { sendEmail() }

        aboutAppBar.addOnOffsetChangedListener(object : AppBarLayout.OnOffsetChangedListener {
            var isShow = false
            var scrollRange = -1

            override fun onOffsetChanged(appBarLayout: AppBarLayout, verticalOffset: Int) {
                if (scrollRange == -1) {
                    scrollRange = appBarLayout.totalScrollRange
                }
                if (scrollRange + verticalOffset == 0) {
                    isShow = true
                } else if (isShow) {
                    isShow = false
                }
            }
        })

        val assetManager = this.applicationContext.assets
        val typeface = Typeface.createFromAsset(assetManager, "fonts/Montserrat-Regular.ttf")

        aboutHeaderDescription.typeface = typeface
        infoTextView.typeface = typeface
        developedByTextView.typeface = typeface
        tryAppTextView.typeface = typeface

        infoCardView.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(DEV_URL)))
        }

        newsAppCardView.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(NEWS_URL)))
        }

        playerAppCardView.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(MUSIC_URL)))
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()

                return true
            }
        }

        return super.onOptionsItemSelected(item)
    }

    private fun sendEmail() {
        val value = arrayOf(getString(R.string.email))
        val emailIntent = Intent(Intent.ACTION_SEND)

        emailIntent.data = Uri.parse("mailto:")
        emailIntent.type = "text/plain"
        emailIntent.putExtra(Intent.EXTRA_EMAIL, value)

        try {
            startActivity(Intent.createChooser(emailIntent, this.resources.getString(R.string.message_choose_title)))
            finish()
        } catch (ex: android.content.ActivityNotFoundException) {
            Toast.makeText(this, resources.getText(R.string.no_email_client), Toast.LENGTH_SHORT).show()
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        overridePendingTransition(R.anim.pull_in_left, R.anim.push_out_right)
    }
}
